src/modules/DASDController.rb

```diff
@@ -851,6 +791,106 @@ def report_error(headline, message, details)
         Yast2::Popup.show(message, headline: headline, details: details)
       end
     end
+
+    # Activates a disk if its not activated
+    #
+    # When the disk is already activated, it returns '8' if the
+    # disk is unformatted or '0' otherwise. The idea is to mimic
+    # the same API that ActivateDisk.
+    #
+    # @return [Boolean] Returns an error code (8 means 'unformatted').
+    def activate_disk_if_needed(channel, diag)
+      disk = find_disks.find { |d| d["channel"] == channel }
+      if disk && active_disk?(disk)
+        log.info "Disk #{channel} is already active. Skipping the activation."
+        return disk["formatted"] ? 0 : 8
+      end
+      ActivateDisk(channel, diag) unless disk && active_disk?(disk)
+    end
+
+    # Determines whether the disk is activated or not
+    #
+    # @param disk [Hash]
+    # @return [Boolean]
+    def active_disk?(disk)
+      io = disk.fetch("resource", {}).fetch("io", []).first
```

Comment:
can you add some documenation here why checking only first io for active is enough? All of them have to be true or false?

Reply:
To be honest, I have not seen any devices with more than one element under the `io` key. Perhaps @wfeldt could shed some light on it (please!).

Reply:
Can happen:

> echo '`Read(.probe.storage)' | /usr/lib/YaST2/bin/y2base stdio scr
> ([])
> ([$["bus":"PCI", "bus_hwcfg":"pci", "class_id":1, "device":"C600/X79 series chipset SATA RAID Controller", "device_id":75814, "driver":"ahci", "driver_module":"ahci", "drivers":[$["active":true, "modprobe":true, "modules":[["ahci", ""]]]], "func_id":2, "modalias":"pci:v00008086d00002826sv00001028sd000005D2bc01sc04i00", "model":"Intel C600/X79 series chipset SATA RAID Controller", "old_unique_key":"ZzFy.wgDWbEqNU+F", "resource":$["io":[$["active":true, "length":8, "mode":"rw", "start":61584], $["active":true, "length":4, "mode":"rw", "start":61568], $["active":true, "length":8, "mode":"rw", "start":61552], $["active":true, "length":4, "mode":"rw", "start":61536], $["active":true, "length":32, "mode":"rw", "start":61472]], "irq":[$["count":261068707, "enabled":true, "irq":40]], "mem":[$["active":true, "length":2048, "start":4213329920]]], "rev":"6", "slot_id":31, "sub_class_id":4, "sub_device_id":67026, "sub_vendor":"Dell", "sub_vendor_id":69672, "sysfs_bus_id":"0000:00:1f.2", "sysfs_id":"/devices/pci0000:00/0000:00:1f.2", "unique_key":"w7Y8.wgDWbEqNU+F", "vendor":"Intel Corporation", "vendor_id":98438], $["bus":"PCI", "bus_hwcfg":"pci", "class_id":1, "device":"C600/X79 series chipset IDE-r Controller", "device_id":73020, "driver":"ata_generic", "driver_module":"ata_generic", "drivers":[$["active":true, "modprobe":true, "modules":[["ata_generic", ""]]], $["active":false, "modprobe":true, "modules":[["pata_acpi", ""]]]], "func_id":2, "modalias":"pci:v00008086d00001D3Csv00001028sd000005D2bc01sc01i85", "model":"Intel C600/X79 series chipset IDE-r Controller", "old_unique_key":"s+ud.E4aVbnRJ1lC", "prog_if":133, "resource":$["io":[$["active":true, "length":8, "mode":"rw", "start":61680], $["active":true, "length":4, "mode":"rw", "start":61664], $["active":true, "length":8, "mode":"rw", "start":61648], $["active":true, "length":4, "mode":"rw", "start":61632], $["active":true, "length":16, "mode":"rw", "start":61616]], "irq":[$["count":0, "enabled":true, "irq":18]]], "rev":"5", "slot_id":22, "sub_class_id":1, "sub_device_id":67026, "sub_vendor":"Dell", "sub_vendor_id":69672, "sysfs_bus_id":"0000:00:16.2", "sysfs_id":"/devices/pci0000:00/0000:00:16.2", "unique_key":"FRkt.E4aVbnRJ1lC", "vendor":"Intel Corporation", "vendor_id":98438]])
((nil))

Reply:
Note sure about DASD, but in general, yes.

Reply:
OK, I have extended the method to consider an array is possible. Thanks!